vllm_hpu_extension/rotary_embed.py

```diff
@@ -53,34 +57,39 @@ def __init__(self,
                                               max_position_embeddings,
                                               base,
                                               is_neox_style,
-                                              dtype=torch.get_default_dtype())
+                                              dtype=self.dtype)
 
     def _compute_inv_freq(self) -> torch.Tensor:
-        inv_freq = 1.0 / (self.base**(
-            torch.arange(0, self.dim, 2).float().to(self.device) / self.dim))
+        inv_freq = 1.0 / (self.base**(torch.arange(
+            0, self.rotary_dim, 2).float().to(self.device) / self.rotary_dim))
         return inv_freq
 
-    def _set_cos_sin_cache(self, seq_len, device, dtype):
+    def _set_cos_sin_cache(self, seq_len):
         self.max_seq_len_cached = seq_len
 
         inv_freq = self._compute_inv_freq()
         t = torch.arange(self.max_seq_len_cached,
-                         device=device,
+                         device=self.device,
                          dtype=inv_freq.dtype)
 
         freqs = torch.einsum("i,j->ij", t, inv_freq)
         # Different from paper, but it uses a different permutation in order
         # to obtain the same calculation
         emb = torch.cat((freqs, freqs), dim=-1)
         self.register_buffer("cos_cached",
-                             emb.cos().to(dtype),
+                             emb.cos().to(self.dtype),
```

Comment:
@SanjuCSudhakaran Are you getting correct values without any NaN for sequence lengths >8K. Eg: 16K, 32K.
I am seeing NaN behavior in OH_Fork. and there freqs = torch.outer(t, self.inv_freq) is being used instead of einsum in [Line No: 75.
](https://github.com/HabanaAI/vllm-hpu-extension/pull/11/files#diff-9d2ba4a4a95eeb3d906653050c955db577068859ab8469874bdcdc0579d20915R75)

Can you confirm so that I can replicate the same in OH-Fork